pages/glossary/default-page-language.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Default Page Language
+key: default-page-language
+unambiguous: true
+objective: true
+input_aspects:
+  - DOM tree
+  - Accessibility tree
+  - CSS Styling
+  - Language
+---
+
+The default page language is the _most common language_ used in a [web page][]. If there are no _words_ in the page, the default language is undefined.
+
+The _most common language_ is determined by counting the number of _words_ in the [web page][] that are part of any of the languages in the [language subtag registry][]. The same word can be part of multiple languages. If the number of words is the same, the default language is undefined.
+
+The _words_ used in the following texts should be counted for the _most common language_:
+
+- **page title**: The value of the [document title][] of the [document][] in the [top-level browsing context][].
+- **default language content**: The value of any [text nodes][] and the [accessible name][] of any [elements][] for which all of the following is true:
+  - The [node][] is [included in the accessibility tree][]; and
```

Comment:
Why including accessible but invisible nodes and not visible but not accessible nodes?

Reply:
Because those nodes will not be processed by assistive technologies. The language of content in aria-hidden doesn't much matter, since it's not going to get read out.

Reply:
Understanding 3.1.1 also list benefits for the visual parts:
> Visual browsers can display characters and scripts correctly. Media players can show captions correctly. 

And so does 3.1.2:
> This is especially important when switching between languages that read from left to right and languages that read from right to left, or when text is rendered in a language that uses a different alphabet. Users with disabilities who know all the languages used in the Web page will be better able to understand the content when each passage is rendered appropriately.

This seems to indicate that a correctly set programmatic language is important both for screen readers and visual display, hence that "visible only" words should apply.


I can imagine a situation where the visual part of the page has been translated, but the accessible hasn't. This would result in the language this definition finds being different whether we count "visible only" and "AT only" words and thus may the rule fail where it shouldn't.